Port to NumPy 2 C API: use NPY_ARRAY_* flags and PyArrayObject pointers

--- a/acor/_acor.c
+++ b/acor/_acor.c
@@ -44,7 +44,7 @@
 {
     int i, j, N, ndim, info, maxlag;
     double *data;
-    PyObject *data_array;
+    PyArrayObject *data_array;
 
     /* Return value */
     PyObject *ret;
@@ -57,7 +57,7 @@
         return NULL;
 
     /* Get the data as a numpy array object */
-    data_array  = PyArray_FROM_OTF(data_obj, NPY_DOUBLE, NPY_IN_ARRAY);
+    data_array  = (PyArrayObject *)PyArray_FROM_OTF(data_obj, NPY_DOUBLE, NPY_ARRAY_IN_ARRAY);
     if (data_array == NULL) {
         PyErr_SetString(PyExc_TypeError, "The input data must be a numpy.ndarray.");
         Py_XDECREF(data_array);
@@ -136,8 +136,8 @@
     int ndim, ndim_out, N, N_out;
     double *data, *out_data;
     double mean;
-    PyObject *data_array;
-    PyObject *out_array;
+    PyArrayObject *data_array;
+    PyArrayObject *out_array;
 
     /* Parse the input tuple */
     PyObject *data_obj, *out_obj;
@@ -145,8 +145,8 @@
         return NULL;
 
     /* Get the data as a numpy array object */
-    data_array = PyArray_FROM_OTF(data_obj, NPY_DOUBLE, NPY_IN_ARRAY);
-    out_array = PyArray_FROM_OTF(out_obj, NPY_DOUBLE, NPY_OUT_ARRAY);
+    data_array = (PyArrayObject *)PyArray_FROM_OTF(data_obj, NPY_DOUBLE, NPY_ARRAY_IN_ARRAY);
+    out_array = (PyArrayObject *)PyArray_FROM_OTF(out_obj, NPY_DOUBLE, NPY_ARRAY_OUT_ARRAY);
     if (data_array == NULL || out_array == NULL) {
         PyErr_SetString(PyExc_TypeError,
                 "The input data must be a numpy.ndarrays.");
